scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -261,7 +261,7 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride with Hf
         // This case is going to result in an error, but we don't want to throw until
         // validateTaps, so we just put a dummy path to return something so the
         // Job constructor does not fail.
-        CastHfsTap(createHfsTap(hdfsScheme, hdfsPaths.head, sinkMode))
+        new InvalidSourceTap(goodHdfsPaths(hdfsMode))
```

Comment:
goodHdfsPaths will return Nil in this case right? We need to pass hdfsPaths here I think so that openForRead throws with the right message. Can you also update the comment above.